sklearn/impute/_base.py

```diff
@@ -64,7 +64,56 @@ def _most_frequent(array, extra_value, n_repeat):
             return extra_value
 
 
-class SimpleImputer(TransformerMixin, BaseEstimator):
+class _BaseImputer(TransformerMixin, BaseEstimator):
+    """Base class for all imputers.
+
+    It add automatically support for `add_indicator`.
+    """
+
+    def __init__(self, missing_values=np.nan, add_indicator=False):
+        self.missing_values = missing_values
+        self.add_indicator = add_indicator
+
+    def _fit_indicator(self, X):
+        """Fit a MissingIndicator and trigger a copy of X as well."""
+        if self.add_indicator:
+            self.indicator_ = MissingIndicator(
+                missing_values=self.missing_values, error_on_new=False
+            )
+            self.indicator_.fit(X)
+        else:
+            self.indicator_ = None
+
+    def _transform_indicator(self, X):
+        """Compute the indicator mask."""
+        if self.add_indicator:
+            if not hasattr(self, 'indicator_'):
+                raise ValueError(
+                    "Make sure to call _fit_indicator before "
+                    "_transform_indicator"
+                )
+            return self.indicator_.transform(X)
+
+    def _concatenate_indicator(self, X_imputed, X_indicator=None):
+        """Concatenate indicator mask with the imputed data."""
+        if self.add_indicator:
+            hstack = sparse.hstack if sparse.issparse(X_imputed) else np.hstack
+            if X_indicator is not None:
+                X_imputed = hstack((X_imputed, X_indicator))
+            else:
+                raise ValueError(
+                    "Impossible to concatenate data from the missing "
```

Comment:
Same here, should we error for a private api?

Reply:
I would argue the same than earlier